src/components/refactor/dialogs/two-windings-transformer-creation/tap-changer-pane/tap-changer-pane-taps.js

```diff
@@ -0,0 +1,368 @@
+                    : null;
+            setValue(`${tapChanger}.${HIGH_TAP_POSITION}`, newHighTapPosition);
+        },
+        [tapChanger, getValues, setValue]
+    );
+
+    // Adjust high tap position when low tap position change + remove red if value fixed
+    useEffect(() => {
+        trigger(`${tapChanger}.${LOW_TAP_POSITION}`).then((result) => {
+            if (result) {
+                resetTapNumbers(null);
+            }
+        });
+    }, [
+        trigger,
+        tapChanger,
+        lowTapPosition, // the only value supposed to change
+        resetTapNumbers,
+    ]);
+
+    // when we detect a change in tapFields (so when the size or the order of the list of rows change), we reset the tap fields
+    useEffect(() => {
+        resetTapNumbers(tapFields);
+    }, [tapFields, resetTapNumbers]);
+
+    const handleCreateTapRule = (lowTap, highTap) => {
+        const currentTapRows = getValues(`${tapChanger}.${STEPS}`);
+
+        if (currentTapRows.length > 1) {
+            let interval = (highTap - lowTap) / (currentTapRows.length - 1);
+            let current = lowTap;
+
+            currentTapRows.forEach((row, index) => {
+                currentTapRows[index][createTapRuleColumn] = current;
+
+                current += interval;
+            });
+            replace(currentTapRows);
+        }
+    };
+
+    function handleImportTapRuleButton() {
+        trigger(`${tapChanger}.${LOW_TAP_POSITION}`).then((result) => {
+            if (result) {
+                setOpenImportRuleDialog(true);
+            }
+        });
+    }
+
+    const handleImportTapRule = (selectedFile, setFileParseError) => {
+        Papa.parse(selectedFile, {
+            header: true,
+            skipEmptyLines: true,
+            complete: function (results) {
+                if (results.data.length > MAX_TAP_NUMBER) {
+                    setFileParseError(
+                        intl.formatMessage(
+                            { id: 'TapPositionValueError' },
+                            { value: MAX_TAP_NUMBER }
+                        )
+                    );
+                    return;
+                }
+                let rows = results.data.map((val) => ({
+                    ...handleImportRow(val),
+                    [SELECTED]: false,
+                }));
+                if (rows && rows.length > 0) {
+                    replace(rows);
+                }
+            },
+        });
+    };
+
+    const lowTapPositionField = (
+        <IntegerInput
+            name={`${tapChanger}.${LOW_TAP_POSITION}`}
+            label="LowTapPosition"
+            formProps={{
+                disabled: disabled,
+            }}
+        />
+    );
+
+    const highTapPositionField = (
+        <IntegerInput
+            name={`${tapChanger}.${HIGH_TAP_POSITION}`}
+            label="HighTapPosition"
+            formProps={{
+                disabled: true,
```

Comment:
Maybe we should use 
```
InputProps={{
            readOnly: true,
}}
```
instead of the disabled formProps as explain in the documentation 
https://mui.com/material-ui/react-text-field/#form-props

disable and read only doesn't have the same meaning.

Reply:
Yeah I tried but it's not the expected UI (PO wants grey fields exactly like if it was disabled), and I didn't want to do some css just for this, but you're right in term of code is better to use this prop. I will change it and see if I need to do some style above or not.

Reply:
I checked with the PO and she definitely want it disabled because the read only prop is not really clear in term of UI (I agree with that). I'm not sure using readOnly and copy the style from disable worth it.